src/Host/Client/Impl/Session/RSessionProvider.cs

```diff
@@ -194,33 +197,50 @@ public class RSessionProvider : IRSessionProvider {
                 }
 
                 OnBrokerStateChanged(new HostLoad());
-                BrokerChanged?.Invoke(this, new EventArgs());
+                OnBrokerChanged();
                 return true;
             }
         }
 
         private async Task SwitchBrokerAsync(CancellationToken cancellationToken) {
-            var transactions = _sessions.Values.Select(s => s.StartSwitchingBroker()).ExcludeDefault().ToList();
+            var transactions = new List<IRSessionSwitchBrokerTransaction>();
+            var sessionsToStop = new List<IRSession>();
+
+            foreach (var session in _sessions.Values) {
+                var transaction = session.StartSwitchingBroker();
+                if (transaction != null) {
+                    transactions.Add(transaction);
+                } else {
+                    sessionsToStop.Add(session);
+                }
+            }
+
             if (transactions.Any()) {
-                await SwitchSessionsAsync(transactions, cancellationToken);
+                await SwitchSessionsAsync(transactions, sessionsToStop, cancellationToken);
```

Comment:
So if we are only to switch interactive session - do we still need transactions and multiple sessions support?